Ability to use anchors with dots in cross references

I've a document like
```
[[os.keystore.create]]
===== keystore create 
```

and would like to link to it from another file
```adoc
<<file1.adoc#os.keystore.create>>
xref:file1.adoc#os.keystore.create[]
```

But both fail because it has problems with file name resolution. As you can see it resolves `file1.adoc#os.keystore.create` with `/var/asciidoc-root-folder/file1.html#os.keystore.adoc`
````
Exception in thread "main" java.lang.RuntimeException: failed to create confluence page for asciidoc content in '/var/asciidoc-root-folder/index.adoc '
        at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.newAsciidocConfluencePage(AsciidocConfluencePage.java:164)
        at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluenceConverter.lambda$buildPageTree$0(AsciidocConfluenceConverter.java:96)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
        at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluenceConverter.buildPageTree(AsciidocConfluenceConverter.java:92)
        at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluenceConverter.convert(AsciidocConfluenceConverter.java:79)
        at org.sahli.asciidoc.confluence.publisher.cli.AsciidocConfluencePublisherCommandLineClient.main(AsciidocConfluencePublisherCommandLineClient.java:103)
Caused by: java.lang.RuntimeException: unable to find cross-referenced page '/var/asciidoc-root-folder/file1.html#os.keystore.adoc'
        at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.lambda$replaceCrossReferenceTargets$9(AsciidocConfluencePage.java:280)
        at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.replaceAll(AsciidocConfluencePage.java:203)
        at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.lambda$replaceCrossReferenceTargets$10(AsciidocConfluencePage.java:259)
        at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.lambda$postProcessContent$4(AsciidocConfluencePage.java:213)
        at java.base/java.util.stream.ReduceOps$1ReducingSink.accept(ReduceOps.java:80)
        at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
        at java.base/java.util.stream.ReferencePipeline.reduce(ReferencePipeline.java:563)
        at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.postProcessContent(AsciidocConfluencePage.java:213)
        at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.convertedContent(AsciidocConfluencePage.java:174)
        at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.newAsciidocConfluencePage(AsciidocConfluencePage.java:158)
        ... 5 more
Caused by: java.io.FileNotFoundException: /var/asciidoc-root-folder/file1.html#os.keystore.adoc (No such file or directory)
        at java.base/java.io.FileInputStream.open0(Native Method)
        at java.base/java.io.FileInputStream.open(FileInputStream.java:219)
        at java.base/java.io.FileInputStream.<init>(FileInputStream.java:157)
        at org.sahli.asciidoc.confluence.publisher.converter.AsciidocConfluencePage.lambda$replaceCrossReferenceTargets$9(AsciidocConfluencePage.java:266)
        ... 18 more
```

--- a/asciidoc-confluence-publisher-converter/src/main/java/org/sahli/asciidoc/confluence/publisher/converter/AsciidocConfluencePage.java
+++ b/asciidoc-confluence-publisher-converter/src/main/java/org/sahli/asciidoc/confluence/publisher/converter/AsciidocConfluencePage.java
@@ -258,7 +258,8 @@ public class AsciidocConfluencePage {
     private static Function<String, String> replaceCrossReferenceTargets(Path pagePath, Map<String, Object> userAttributes, PageTitlePostProcessor pageTitlePostProcessor, Charset sourceEncoding, String spaceKey, Path templatesDir, Path pageAssetsFolder, Path rootFolder) {
         return (content) -> replaceAll(content, PAGE_TITLE_PATTERN, (matchResult) -> {
             String htmlTarget = matchResult.group(1);
-            Path referencedPagePath = pagePath.getParent().resolve(Paths.get(htmlTarget.substring(0, htmlTarget.lastIndexOf('.')) + ".adoc"));
+            String htmlTargetWithoutAnchor = htmlTarget.contains("#") ? htmlTarget.substring(0, htmlTarget.indexOf('#')) : htmlTarget;
+            Path referencedPagePath = pagePath.getParent().resolve(Paths.get(htmlTargetWithoutAnchor.substring(0, htmlTargetWithoutAnchor.lastIndexOf('.')) + ".adoc"));
 
             try {
                 Map<String, Object> userAttributesWithMaskedNullValues = maskNullWithEmptyString(userAttributes);

--- a/asciidoc-confluence-publisher-converter/src/test/java/org/sahli/asciidoc/confluence/publisher/converter/AsciidocConfluencePageTest.java
+++ b/asciidoc-confluence-publisher-converter/src/test/java/org/sahli/asciidoc/confluence/publisher/converter/AsciidocConfluencePageTest.java
@@ -1274,6 +1274,22 @@ public class AsciidocConfluencePageTest {
     }
 
     @Test
+    public void renderConfluencePage_asciiDocWithInterDocumentCrossReferenceAnchorContainingDots_returnsConfluencePageWithLinkToReferencedPageByPageTitleWithAnchor() {
+        // arrange
+        Path rootFolder = copyAsciidocSourceToTemporaryFolder("src/test/resources/inter-document-cross-references");
+        AsciidocPage asciidocPage = asciidocPage(rootFolder, "source-page-reference-with-dotted-anchor.adoc");
+
+        // act
+        AsciidocConfluencePage asciidocConfluencePage = newAsciidocConfluencePage(asciidocPage, UTF_8, TEMPLATES_FOLDER, assetsTargetFolderFor(asciidocPage), "TEST");
+
+        // assert
+        String expectedContent = "<p>This is a <ac:link ac:anchor=\"os.keystore.create\"><ri:page ri:content-title=\"Target Page\" ri:space-key=\"TEST\"></ri:page>" +
+                "<ac:plain-text-link-body><![CDATA[reference with dotted anchor]]></ac:plain-text-link-body>" +
+                "</ac:link> to the target page.</p>";
+        assertThat(asciidocConfluencePage.content(), is(expectedContent));
+    }
+
+    @Test
     public void renderConfluencePage_asciiDocWithInterDocumentCrossReferenceToTargetPageWithHtmlCharacterInTitle_returnsConfluencePageWithLinkToReferencedPageByPageTitle() {
         // arrange
         Path rootFolder = copyAsciidocSourceToTemporaryFolder("src/test/resources/inter-document-cross-references");

--- a/asciidoc-confluence-publisher-converter/src/test/resources/inter-document-cross-references/source-page-reference-with-dotted-anchor.adoc
+++ b/asciidoc-confluence-publisher-converter/src/test/resources/inter-document-cross-references/source-page-reference-with-dotted-anchor.adoc
@@ -1,0 +1,3 @@
+= Source Page
+
+This is a <<target-page-with-anchor.adoc#os.keystore.create,reference with dotted anchor>> to the target page.

--- a/asciidoc-confluence-publisher-converter/src/test/resources/inter-document-cross-references/target-page-with-anchor.adoc
+++ b/asciidoc-confluence-publisher-converter/src/test/resources/inter-document-cross-references/target-page-with-anchor.adoc
@@ -4,3 +4,6 @@ This is the target page with an:
 
 [[_anchor]]
 Anchor
+
+[[os.keystore.create]]
+Anchor with dots

